### PR TITLE
Drop Apache 2.0 compatibility code

### PR DESCRIPTION
**Pull request by ekohl**
It's safe to assume the version is >= 2.2 so the conditional can be dropped.

### CODE REVIEW

**Review by bastelfreak — APPROVED** [review]

File: manifests/mod/proxy_connect.pp [hunk 1]
```diff
@@ -2,17 +2,14 @@
 #   Installs `mod_proxy_connect`.
 # 
 # @param apache_version
-#   Used to verify that the Apache version you have requested is compatible with the module.
+#   Deprecated and unused. Remains here for compatibility.
 # 
 # @see https://httpd.apache.org/docs/current/mod/mod_proxy_connect.html for additional documentation.
 #
 class apache::mod::proxy_connect (
   $apache_version  = undef,
```
Comment (ekohl) [inline manifests/mod/proxy_connect.pp:10]: I kept this variable in to not make it a breaking change. Silently ignoring it is safe, but it should be removed with the next major release.

Comment (bastelfreak) [reply manifests/mod/proxy_connect.pp:10]: should we open a PR that drops it? I'm quite sure otherwise we will forget it.

**Review by bastelfreak — DISMISSED** [review]

File: manifests/mod/proxy_connect.pp (hunk 1 above)
Comment (ekohl) [reply manifests/mod/proxy_connect.pp:10]: I thought about the same thing.

**Review by smortex — APPROVED** [review]

**Comment by chelnak** [comment]
@david22swan This looks good to go to me but I'm wondering if we should include the breaking change mentioned by @ekohl above. There is no real reason we can't do a major release.

What do you think?

**Comment by david22swan** [comment]
@ekohl @chelnak Seeing as we have a breaking change already waiting for a release with the removal of OS compatibility code I'd say go for it and remove it.

**Comment by chelnak** [comment]
I'm down with that.

@ekohl Feel free to remove the variable in this PR 😄

**Comment by ekohl** [comment]
Updated to drop the parameter.

**Review by bastelfreak — APPROVED** [review]

**Review by david22swan — APPROVED** [review]

**Comment by david22swan** [comment]
@ekohl Thanks again for the work :)

**Pull request merged**